shared/processing/dataGetters.ts

```diff
@@ -0,0 +1,319 @@
+        processItemsInto<IDescriptionComponents>(getDescriptionItemsData)(category.description, dataBlob, linkedItems);
+
+        const dataObject: ICategory = {
+            ...getSystemProperties(category),
+            apiReference: processTaxonomyElement(category.apiReference),
+            description: category.description.getHtml(),
+            name: category.name.value,
+            url: category.url.value,
+        };
+
+        categories.push(getWrappedData<ICategory>(dataObject, category));
+    });
+
+    return categories;
+};
+
+export const getPathOperationsData = (
+    items: ZapiPathOperation[],
+    dataBlob: IPreprocessedData,
+    linkedItems: ContentItem[],
+): Array<IWrappedItem<IPathOperation>> => {
+    const paths = [];
+
+    items.map((pathOperation: ZapiPathOperation) => {
+        // Category items already processed within the zAPI Specification object
+        processItemsInto<IParameter>(getParametersData)(pathOperation.parameters, dataBlob, linkedItems);
+        processItemsInto<IRequestBody>(getRequestBodiesData)(pathOperation.requestBody, dataBlob, linkedItems);
+        processItemsInto<IResponse>(getResponsesData)(pathOperation.responses, dataBlob, linkedItems);
+        processItemsInto<IDescriptionComponents>(getDescriptionItemsData)(
+            pathOperation.description,
+            dataBlob,
+            linkedItems,
+        );
+
+        const dataObject: IPathOperation = {
+            ...getSystemProperties(pathOperation),
+            apiReference: processTaxonomyElement(pathOperation.apiReference),
+            category: processLinkedItemsElement(pathOperation.category),
+            codeSamples: processLinkedItemsElement(pathOperation.codeSamples),
+            deprecated: processMultipleChoiceElement(pathOperation.deprecated),
+            description: pathOperation.description.getHtml(),
+            name: pathOperation.name.value,
+            parameters: processLinkedItemsElement(pathOperation.parameters),
+            path: pathOperation.path.value,
+            pathOperation: processTaxonomyElement(pathOperation.pathOperation),
+            requestBody: pathOperation.requestBody.getHtml(),
+            responses: pathOperation.responses.getHtml(),
+            url: pathOperation.url.value,
+        };
+
+        paths.push(getWrappedData(dataObject, pathOperation));
+    });
+
+    return paths;
+};
+
+export const getServersData = (
+    field: RichTextField,
+    dataBlob: IPreprocessedData,
+    linkedItems: ContentItem[],
+): Array<IWrappedItem<IServer>> => {
+    const servers = [];
+
+    field.linkedItemCodenames.map((codename: string) => {
+        const server = getFromLinkedItems<ZapiServer>(codename, linkedItems);
+        const dataObject: IServer = {
+            ...getSystemProperties(server),
+            description: server.description.value,
+            url: server.url.value,
+        };
+
+        servers.push(getWrappedData<IServer>(dataObject, server));
```

Comment:
Again and again (just want to point out it is related to multiple places, no code bashing intended). Why amending a new array if map returns a value and is perfect fit for x to y transformation (`reduce` can be used for x to y[], but this is not the case)